Hash function's new typing, such as faker.sha1, is not backward compatible

Hi, thank you very much for this library

The version [36.2.1](https://github.com/joke2k/faker/commit/2f20f76db8921cc97b257d04eca75916485028a0) added new overloads so that sha1 is correctly typed related to the raw_output... but the function itself has a default so calling it with no params should be supported.

```
value = faker.sha1()
````
produces
```
error: All overload variants of "sha1" of "Faker" require at least one argument  [call-overload]
note: Possible overload variants:
note:     def sha1(self, raw_output: Literal[True]) -> bytes
note:     def sha1(self, raw_output: Literal[False]) -> str
```

Versions / config
Python: 3.12.8
faker: 36.2.2
mypy: 1.15.0
mypy config:
```
[tool.mypy]
namespace_packages=true
explicit_package_bases=true
strict=true
show_error_codes=true
```

--- a/faker/providers/misc/__init__.py
+++ b/faker/providers/misc/__init__.py
@@ -61,6 +61,9 @@ class Provider(BaseProvider):
         return os.urandom(length)
 
     @overload
+    def md5(self) -> str: ...
+
+    @overload
     def md5(self, raw_output: Literal[True]) -> bytes: ...
 
     @overload
@@ -81,6 +84,9 @@ class Provider(BaseProvider):
         return res.hexdigest()
 
     @overload
+    def sha1(self) -> str: ...
+
+    @overload
     def sha1(self, raw_output: Literal[True]) -> bytes: ...
 
     @overload
@@ -99,6 +105,9 @@ class Provider(BaseProvider):
         if raw_output:
             return res.digest()
         return res.hexdigest()
+
+    @overload
+    def sha256(self) -> str: ...
 
     @overload
     def sha256(self, raw_output: Literal[True]) -> bytes: ...

--- a/faker/proxy.pyi
+++ b/faker/proxy.pyi
@@ -2203,6 +2203,19 @@ class Faker:
         ...
 
     @overload
+    def md5(self) -> str:
+        """
+        Generate a random MD5 hash.
+
+        If ``raw_output`` is ``False`` (default), a hexadecimal string representation of the MD5 hash
+        will be returned. If ``True``, a ``bytes`` object representation will be returned instead.
+
+        :sample: raw_output=False
+        :sample: raw_output=True
+        """
+        ...
+
+    @overload
     def md5(self, raw_output: Literal[True]) -> bytes:
         """
         Generate a random MD5 hash.
@@ -2278,6 +2291,19 @@ class Faker:
         ...
 
     @overload
+    def sha1(self) -> str:
+        """
+        Generate a random SHA-1 hash.
+
+        If ``raw_output`` is ``False`` (default), a hexadecimal string representation of the SHA-1 hash
+        will be returned. If ``True``, a ``bytes`` object representation will be returned instead.
+
+        :sample: raw_output=False
+        :sample: raw_output=True
+        """
+        ...
+
+    @overload
     def sha1(self, raw_output: Literal[True]) -> bytes:
         """
         Generate a random SHA-1 hash.
@@ -2296,6 +2322,19 @@ class Faker:
         Generate a random SHA-1 hash.
 
         If ``raw_output`` is ``False`` (default), a hexadecimal string representation of the SHA-1 hash
+        will be returned. If ``True``, a ``bytes`` object representation will be returned instead.
+
+        :sample: raw_output=False
+        :sample: raw_output=True
+        """
+        ...
+
+    @overload
+    def sha256(self) -> str:
+        """
+        Generate a random SHA-256 hash.
+
+        If ``raw_output`` is ``False`` (default), a hexadecimal string representation of the SHA-256 hash
         will be returned. If ``True``, a ``bytes`` object representation will be returned instead.
 
         :sample: raw_output=False

